src/taipy/core/_scheduler/_dispatcher/_job_dispatcher.py

```diff
@@ -20,19 +24,93 @@
 from ...job._job_manager_factory import _JobManagerFactory
 from ...job.job import Job
 from ...task.task import Task
+from .._abstract_scheduler import _AbstractScheduler
 
 
-class _JobDispatcher:
+class _JobDispatcher(threading.Thread):
     """Manages job dispatching (instances of `Job^` class) on executors."""
 
-    def __init__(self):
-        pass
+    __STOP_FLAG = False
+    _dispatched_processes: Dict = {}
+    __logger = _TaipyLogger._get_logger()
+    lock = Lock()
+
+    def __init__(self, scheduler: _AbstractScheduler):
+        threading.Thread.__init__(self, name="Thread-Taipy-JobDispatcher")
+        self.daemon = True
+        self.scheduler = scheduler
+
+    def start(self):
+        #     """TODO: doc"""
+        return NotImplemented
+
+    def is_running(self) -> bool:
+        #     """TODO: doc"""
+        return NotImplemented
+
+    def stop(self):
+        #     """TODO: doc"""
+        return NotImplemented
+
+    def run(self):
+        while True and not self.__STOP_FLAG:
+            try:
+                if self._can_execute() and (job := self.scheduler.jobs_to_run.get(block=True)):
+                    self._execute_job(job)
+                else:
+                    sleep(0.1)  # Limit CPU usage
+            except:  # In case the last job of the queue has been removed.
+                self.__logger.warning(f"{job.id} is no longer in the list of jobs to run.")
```

Comment:
@jrobinAV so at first, i have only `while True` to keep the loop going, then I added `STOP_FLAG` to make sure we can manually stop the thread. I want to keep this part (running the loop) and the condition of dispatching a job or not separated for readability. Since the addition of `STOP_FLAG`, the while will now be only `while STOP_FLAG` but I'd still rather keep the condition `if self._can_execute() and (job := self.scheduler.jobs_to_run.get(block=True)):` separated as above. It also allows us to make use of sleep to limit the CPU usage and the try catch would make more sense too since the try-catch is for `self.scheduler.jobs_to_run.get`

Reply:
OK that s good for me.